tools/crm_resource.c

```diff
@@ -1726,18 +1754,57 @@ handle_set_param(pcmk_resource_t *rsc, xmlNode *cib_xml_orig)
 }
 
 static int
-handle_wait(void)
+handle_wait(pcmk_resource_t *rsc, pcmk_node_t *node, xmlNode *cib_xml_orig,
+            pcmk_scheduler_t *scheduler)
 {
     return wait_till_stable(out, options.timeout_ms, cib_conn);
 }
 
 static int
-handle_why(pcmk_resource_t *rsc, pcmk_node_t *node)
+handle_why(pcmk_resource_t *rsc, pcmk_node_t *node, xmlNode *cib_xml_orig,
+           pcmk_scheduler_t *scheduler)
 {
     return out->message(out, "resource-reasons-list",
                         scheduler->priv->resources, rsc, node);
 }
 
+typedef int (*crm_resource_fn_t)(pcmk_resource_t *, pcmk_node_t *, xmlNode *,
```

Comment:
More arguments will probably get added to phase out global variables where possible. This is WIP.